src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,172 @@
-        catalog = catalog[separations < radius]
+        separations = stars["ra_dec"].separation(pointing)
+        stars["separation"] = separations
+        stars = stars[separations < radius]
+
+    return stars
+
+
+def get_star_catalog(catalog, min_magnitude=0.0, max_magnitude=10.0, row_limit=1000000):
+    """
+    Utility function to download a star catalog for the get_bright_stars function
+
+    Parameters
+    ----------
+    catalog: string
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalogues. Default: Yale
+    min_magnitude: float
+        Minimum value for magnitude used in lookup
+    max_magnitude: float
+        Maximum value for magnitude used in lookup
+    row_limit: int
+        Maximum number of rows for the star catalog lookup
+
+    Returns
+    ----------
+    Astropy table:
+       List of all stars after cuts with catalog numbers, magnitudes,
+       and coordinates as SkyCoord objects including proper motion
+    """
+    from astroquery.vizier import Vizier
+
+    catalog_dict = StarCatalogues[catalog].value
+
+    vizier = Vizier(
+        catalog=catalog_dict["directory"],
+        columns=catalog_dict["columns"],
+        row_limit=row_limit,
+    )
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    stars = vizier.query_constraints(Vmag="{min_magnitude}..{max_magnitude}")[0]
+
+    stars.meta["Catalog"] = StarCatalogues[catalog].value
+
+    return stars
+
+
+def get_bright_stars(
+    time, catalog="Yale", pointing=None, radius=None, magnitude_cut=None
+):
+    """
+    Get an astropy table of bright stars from the Yale bright star catalog
+
+    Parameters
+    ----------
+    time: astropy Time
+        time to which proper motion is applied
+    catalog : string
+        name of the catalog to be used available catalogues are 'Yale' and 'Hipparcos'. Default: 'Yale'
+    pointing: astropy Skycoord
+       pointing direction in the sky (if none is given, full sky is returned)
+    radius: astropy angular units
+       Radius of the sky region around pointing position. Default: full sky
+    magnitude_cut: float
+        Return only stars above a given absolute magnitude. Default: None (all entries)
+
+    Returns
+    -------
+    Astropy table:
+       List of all stars after cuts with catalog numbers, magnitudes,
+       and coordinates as SkyCoord objects including proper motion
+    """
+
+    from ctapipe.utils import get_table_dataset
+
+    cat = StarCatalogues[catalog].value
+
+    stars = get_table_dataset(cat["record"], role="bright star catalog")
+
+    stars["ra_dec"] = SkyCoord(
+        ra=Angle(stars["RAJ2000"], unit=u.deg),
+        dec=Angle(stars["DEJ2000"], unit=u.deg),
+        pm_ra_cosdec=stars["pmRA"].quantity,  # yes, pmRA is already pm_ra_cosdec
+        pm_dec=stars["pmDE"].quantity,
+        frame="icrs",
+        obstime=Time("J2000.0"),
+    )
+    stars["ra_dec"].apply_space_motion(new_obstime=time)
```

Comment:
This function returns a new coordinate, you need to assign it:

```
stars["ra_dec"] = stars["ra_dec"].apply_space_motion(new_obstime=time)
``` 

Also, please have a look at the astropy issue I linked for the workaround to transforming these coordinantes then to AltAz.

It would also be good to add a test converting the resulting star coordinates to `AltAz` to make sure this works.

https://docs.astropy.org/en/stable/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord.apply_space_motion